google-http-client-apache-v5/src/main/java/com/google/api/client/http/apache/v5/Apache5HttpResponse.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.http.apache.v5;
+
+import com.google.api.client.http.LowLevelHttpResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.message.StatusLine;
+
+final class Apache5HttpResponse extends LowLevelHttpResponse {
+
+  private static final Logger LOGGER = Logger.getLogger(Apache5HttpResponse.class.getName());
+  private final HttpUriRequestBase request;
+  private final ClassicHttpResponse response;
+  private final Header[] allHeaders;
+  private final HttpEntity entity;
+  private final Apache5ResponseContent content;
+
+  Apache5HttpResponse(HttpUriRequestBase request, ClassicHttpResponse response) {
+    this.request = request;
+    this.response = response;
+    this.allHeaders = response.getHeaders();
+    this.entity = response.getEntity();
+    Apache5ResponseContent contentValue = null;
+    if (entity != null) {
+      try {
+        contentValue = new Apache5ResponseContent(entity.getContent(), response);
+      } catch (IOException ex) {
+        LOGGER.log(Level.SEVERE, "Error when obtaining content from the response", ex);
```

Comment:
This swallows the `IOException` rather than throwing it from `getContent()` like we used to.

Creating our wrapper class is very low-cost, so we should be able to just create it when needed -- thus preserving the IOException throwing behavior:
```java
  @Override
  public InputStream getContent() throws IOException {
    return new Apache5ResponseContent(entity.getContent(), response);
  }
```

Reply:
Makes sense, thanks. Done.